src/api/undeploy/helpers/delete-deployment-file.js

```diff
@@ -0,0 +1,21 @@
+import { createLogger } from '~/src/helpers/logging/logger.js'
+import { featureToggles } from '~/src/helpers/feature-toggle/feature-toggles.js'
+import { isFeatureEnabled } from '~/src/helpers/feature-toggle/is-feature-enabled.js'
+import { removeDeployment } from '~/src/helpers/remove/workflows/remove-deployment.js'
+
+/**
+ * @param {{serviceName: string, logger: [import('pino').Logger]}} options
+ */
+export async function deleteDeploymentFiles({
+  serviceName,
+  logger = createLogger()
```

Comment:
IMO this is a bit of a weird pattern. This is being passed in for ease of testing? Why not just mock the logger in tests? Then you don't have "test code" in production code. Which I'm not a massive fan of

Reply:
If I don't pass the logger the statements from different async calls tend to be out of sync and harder to trace the journey. Also this seems a lot tidier.